File: src/1394.find-lucky-integer-in-an-array.1510/solution.js
/**
 * @param {number[]} arr
 * @return {number}
 */
var findLucky = function (arr) {
    const countBy = arr.reduce((obj, item) => {
        obj[item] = (obj[item] || 0) + 1;
        return obj;
    }, {});
    let result = -1;
    Object.entries(countBy).forEach(([
        key, value, ]) => {
        if (+key === value && value > result) {
            result = value;
        }
    });
    return result;
};

File: src/1394.find-lucky-integer-in-an-array.1510/solution.java
class Solution {
    public int findLucky(int[] arr) {
        int[] counts = new int[arr.length+1];
        for(int n : arr){
            if(n>arr.length){
                continue;
            }
            counts[n]++;
        }
        for(int i=counts.length-1;i>0;i--){
            if(counts[i] == i){
                return i;
            }
        }
        return -1;
    }
}